beacon/engine/types.go

```diff
@@ -260,7 +260,15 @@ func ExecutableDataToBlockNoHash(data ExecutableData, versionedHashes []common.H
 
 	var requestsHash *common.Hash
 	if requests != nil {
-		h := types.CalcRequestsHash(requests)
+		// Put back request type byte.
+		typedRequests := make([][]byte, len(requests))
+		for i, reqdata := range requests {
+			typedReqdata := make([]byte, len(reqdata)+1)
+			typedReqdata[0] = byte(i)
+			copy(typedReqdata[1:], reqdata)
+			typedRequests[i] = typedReqdata
+		}
+		h := types.CalcRequestsHash(typedRequests)
```

Comment:
How about this?

```go
func CalcRequestsHash(requests [][]byte, hasTypePrefix bool) common.Hash {
	h1, h2 := sha256.New(), sha256.New()
	var buf common.Hash
	for idx, item := range requests {
		h1.Reset()
		if !hasTypePrefix {
			h1.Write([]byte{byte(idx)})
		}
		h1.Write(item)
		h2.Write(h1.Sum(buf[:0]))
	}
	h2.Sum(buf[:0])
	return buf
}
```

Reply:
I'm using this method of stripping/unstripping the type byte because I expect to remove it again in the next iteration.